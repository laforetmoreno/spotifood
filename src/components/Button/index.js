import React from 'react';
import { string } from 'prop-types';

import styles from './index.module.scss';

const Button = ({ children, className, link, target }) => {
  if (link) {
    return (
      <a
        target={target}
        without
        rel="noopener noreferrer"
        className={(className, styles.link)}
        href={link}
      >
        {children}
      </a>
    );
  }
  return (
    <button type="button" className={(className, styles.wrapper)}>
      {children}
    </button>
  );
};

Button.propTypes = {
  className: string,
  children: string,
  link: string,
  target: string,
};

Button.defaultProps = {
  className: '',
  children: '',
  link: '',
  target: '_blank',
};

export default Button;
