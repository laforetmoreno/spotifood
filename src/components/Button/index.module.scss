@import '../../styles/mixins';

.wrapper {
  border: none;
}

.link {
  @include flex-default();
  @include flex-column();
  @include reset-link();
}
