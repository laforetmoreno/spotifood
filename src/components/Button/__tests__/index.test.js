import React from 'react';
import { render, waitForElement, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Button from '../index';

describe('<Button />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Button>Click here</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a submit button', () => {
    const tree = renderer.create(<Button type="submit">Click here</Button>);
    expect(tree).toMatchSnapshot();
  });

  it('get by id and click', async () => {
    const { getByTestId } = render(<Button />);
    const fieldNode = await waitForElement(() => getByTestId('btn'));
    fireEvent.click(fieldNode);
  });

  it('Unmount Component At Node ', () => {
    let container = null;
    beforeEach(() => {
      container = document.createElement(<Button />);
      document.body.appendChild(container);
    });

    afterEach(() => {
      unmountComponentAtNode(container);
      container.remove();
      container = null;
    });
  });
});
