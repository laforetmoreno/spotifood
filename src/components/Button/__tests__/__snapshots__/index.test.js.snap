// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<button
  className="wrapper"
  data-testid="btn"
  onClick={[Function]}
  type="button"
>
  Click here
</button>
`;

exports[`<Button /> should render a submit button 1`] = `
<button
  className="wrapper"
  data-testid="btn"
  onClick={[Function]}
  type="button"
>
  Click here
</button>
`;
